src/kohlrahbi/ahbtable/ahbpackagetable.py

```diff
@@ -0,0 +1,149 @@
+from pydantic import BaseModel, ConfigDict
+
+from kohlrahbi.logger import logger
+
+# pylint: disable=duplicate-code
+
+
+class AhbPackageTable(BaseModel):
+    """
+    This class contains the AHB Package table as you see it in the beginning AHB documents,
+    but in a machine readable format. Caution: if two PackageTables objects are combined
+    so far only the package_dict field is updated.
+    """
+
+    table: pd.DataFrame = pd.DataFrame()
+    package_dict: dict[EdifactFormat, dict[str, str]] = {}
+    model_config = ConfigDict(arbitrary_types_allowed=True)
+
+    @classmethod
+    def from_docx_table(cls, docx_tables: list[DocxTable]) -> "AhbPackageTable":
+        """
+        Create an AhbPackageTable object from a docx table.
+
+        """
+        table_data = []
+        for table in docx_tables:
+            for row in table.rows:
+                row_data = [cell.text for cell in row.cells]
+                table_data.append(row_data)
+
+        headers = table_data[0]
+        data = table_data[1:]
+        df = pd.DataFrame(data, columns=headers)
+        return cls(table=df)
+
+    def provide_conditions(self, edifact_format: EdifactFormat) -> dict[EdifactFormat, dict[str, str]]:
+        """collect conditions from package table and store them in conditions dict."""
+        conditions_dict: dict[EdifactFormat, dict[str, str]] = {edifact_format: {}}
+
+        df = self.table
+        there_are_conditions = (df["Bedingungen"] != "").any()
+        if there_are_conditions:
+            for conditions_text in df["Bedingungen"][df["Bedingungen"] != ""]:
+                # Split the input into parts enclosed in square brackets and other parts
+                matches = re.findall(
+                    r"\[(\d+)](.*?)(?=\[\d+]|$)",
+                    conditions_text,
+                    re.DOTALL,
+                )
+                for match in matches:
+                    # make text prettier:
+                    text = match[1].strip()
+                    text = re.sub(r"\s+", " ", text)
+
+                    # check whether condition was already collected:
+                    existing_text = conditions_dict[edifact_format].get(match[0])
+                    is_condition_key_collected_yet = existing_text is not None
+                    if is_condition_key_collected_yet:
+                        key_exits_but_shorter_text = len(text) > len(edifact_format)
+                    if not is_condition_key_collected_yet or key_exits_but_shorter_text:
+                        conditions_dict[edifact_format][match[0]] = text
+
+        logger.info("The package conditions for %s were collected.", edifact_format)
+        return conditions_dict
+
+    def provide_packages(self, edifact_format: EdifactFormat):
+        """collect conditions from package table and store them in conditions dict."""
+        package_dict: dict[EdifactFormat, dict[str, str]] = {edifact_format: {}}
+
+        df = self.table
```

Comment:
what is this for? Can we not use `self.table` directly?

Reply:
ddd45ed66730feace8a4f6f06b6def670ed7ec77

Reply:
The commit is not linked
Did you push your changes?

Reply:
The link in the comment is not created 🫣
ddd45ed66730feace8a4f6f06b6def670ed7ec77